module1/tasks.js

```diff
@@ -0,0 +1,189 @@
+}
+function convertChar(c) {
+  return isUpperCase(c) ? c.toLowerCase() : c.toUpperCase();
+}
+
+const changeCase2 = (str) => {
+  return [...str]  //transform string to array of chars
+    .map(c => convertChar(c))
+    .join("");  //collect
+};
+
+
+describe('changeCase', () => {
+  it('should change the capitalization of all letters in a given string', () => {
+    expect(changeCase('21century')).to.equal('21CENTURY');
+    expect(changeCase('Hybris')).to.equal('hYBRIS');
+    expect(changeCase2('21century')).to.equal('21CENTURY');
+    expect(changeCase2('Hybris')).to.equal('hYBRIS');
+  });  
+});
+
+// ==== Filter out the non-unique values in an array ====
+const filterNonUnique = (arr) => {
+  let map = new Map();
+  for(let elem of arr) {
+    if(map.has(elem)) {
+      map.set(elem, true);
+    } else {
+      map.set(elem, false);
+    }
+  }
+  return [...map]
+    .filter( ([key, value]) => value == false )
+    .map(([key, value]) => key);
+};
+
+describe('filterNonUnique', () => {
+  it('should filter out non-unique values in an array', () => {
+    expect(filterNonUnique([1, 2, 2, 3, 4, 4, 5])).to.eql([1,3,5]);
+    expect(filterNonUnique([1, 2, 3, 4])).to.eql([1,2,3,4]);
+  });
+});
+
+// ==== Sort string in alphabetical order ====
+const alphabetSort = (str) => {return[...str].sort().join("")};
+
+describe('alphabetSort', () => {
+  it('should accept a string type only', () => {
+    expect(() => alphabetSort()).to.throw();
+    expect(() => alphabetSort('Text')).not.to.throw();
+  });
+
+  it('should convert the letters of a given string in alphabetical order', () => {
+    expect(alphabetSort('Python')).to.equal('Phnoty');
+  });
+});
+
+// ==== Get min integer ====
+const getSecondMinimum = (arr) => {return arr.sort()[1]};
+
+describe('getSecondMinimum', () => {
+  it('should get array of integers and return second minimum value', () => {
+    expect(getSecondMinimum([5,0,7,3,8])).to.equal(3);
+  });
+});
+
+// ==== Double every even integer ====
+const doubleEveryEven = (arr) => { 
+  return arr.map(e => e % 2 == 0 ? e*2 : e);
+};
+
+describe('doubleEveryEven', () => {
+  it('should get array of integers and return another array of integers where every even number is doubled', () => {
+    expect(doubleEveryEven([2,0,7,3,8,4])).to.eql([4,0,7,3,16,8]);
+  });
+});
+
+// ==== Create array with all possible pairs of two arrays ====
+const getArrayElementsPairs = (a1, a2) => {
+  let result = new Array();
+  for(let e1 of a1) {
+    for(let e2 of a2) {
+      result.push([e1, e2]);
+    }
+  }
+  return result;
+};
+
+describe('getArrayElementsPairs', () => {
+  it('should get two arrays and return array containing each possible pair from the arrays', () => {
+    expect(getArrayElementsPairs([1,2], ['a', 'b'])).to.eql([[1, "a"], [1, "b"], [2, "a"], [2, "b"]]);
+  });
+});
+
+// ==== Deep equal ====
+const deepEqual = (a1 , a2) => {  
+  if(Object.entries(a1).length != Object.entries(a2).length) {
```

Comment:
Please, read the difference between `!=` and `!== `and avoid first one. To be honest, I've never met a situation where I need `!=`.